crates/oxc_linter/src/rules/vitest/prefer_spy_on.rs

```diff
@@ -0,0 +1,324 @@
+        return false;
+    };
+
+    let Some(property_name) = static_member.static_property_name() else {
+        return false;
+    };
+
+    property_name == "mockImplementation"
+}
+
+fn auto_fix_mock_implementation<'a>(
+    right_assignment: &'a Expression<'a>,
+    fn_call_node: &'a Expression<'a>,
+    ctx: &LintContext<'a>,
+) -> String {
+    if parent_has_mock_implementation_call(right_assignment) {
+        return String::new();
+    }
+
+    match fn_call_node {
+        Expression::CallExpression(call) => {
+            let Some(arguments_span) = call.arguments_span() else {
+                return ".mockImplementation()".to_string();
+            };
+
+            return format!(".mockImplementation({})", ctx.source_range(arguments_span));
+        }
+        _ => String::new(),
+    }
+}
+
+impl Rule for PreferSpyOn {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        if let AstKind::AssignmentExpression(assignment) = node.kind() {
+            if !assignment.left.is_member_expression() {
+                return;
+            }
+
+            let result = get_vitest_fn_call(&assignment.right);
+
+            if let Some(fn_call_node) = result {
+                ctx.diagnostic_with_suggestion(
+                    prefer_spy_on_diagnostic(assignment.span),
+                    |fixer| {
+                        let mock_implementation =
+                            auto_fix_mock_implementation(&assignment.right, &fn_call_node, &ctx);
+                        let left_expression = assignment.left.to_member_expression();
+                        let fixer = fixer.for_multifix();
+                        let mut rule_fixes = fixer.new_fix_with_capacity(3);
+
+                        match left_expression {
+                            MemberExpression::ComputedMemberExpression(left)
+                                if left.object.is_identifier_reference() =>
+                            {
+                                rule_fixes
+                                    .push(fixer.insert_text_before_range(left.span, "vi.spyOn("));
+
+                                let identifier = {
+                                    match &left.object {
+                                        Expression::Identifier(id) => id.span,
+                                        _ => Span::empty(0),
+                                    }
+                                };
+
+                                let property = {
+                                    match &left.expression {
+                                        Expression::StringLiteral(property) => property.span,
+                                        Expression::TemplateLiteral(property) => property.span,
+                                        Expression::BinaryExpression(property) => property.span,
+                                        Expression::Identifier(property) => property.span,
+                                        _ => Span::empty(0),
```

Comment:
Here I'm not sure If I have to move the `rules_fixes.push` to the end and do an early return.